File: java-api-core/src/com/kinvey/java/core/KinveyCachedAggregateCallback.java
package com.kinvey.java.core;

import com.kinvey.java.model.Aggregation;

import java.util.List;

/**
 * Created by yuliya on 10/06/17.
 */

public abstract class KinveyCachedAggregateCallback implements KinveyClientCallback<List<Aggregation.Result>> {

    @Override
    public void onSuccess(List<Aggregation.Result> result) {
        Aggregation response = new Aggregation(result);
        onSuccess(response);
    }


    public abstract void onFailure(Throwable error);

    public abstract void onSuccess(Aggregation response);

}

File: java-api-core/src/com/kinvey/java/core/KinveyCachedAggregateCallback.kt
package com.kinvey.java.core

import com.kinvey.java.model.Aggregation

/**
 * Created by yuliya on 10/06/17.
 */

abstract class KinveyCachedAggregateCallback : KinveyClientCallback<List<Aggregation.Result>> {

    override fun onSuccess(result: List<Aggregation.Result>) {
        val response = Aggregation(result)
        onSuccess(response)
    }

    abstract override fun onFailure(error: Throwable)

    abstract fun onSuccess(response: Aggregation)
}
